Automattic/wp-calypso
# Handle press this functionality in the wp-admin editor.

Fixes https://github.com/Automattic/wp-calypso/issues/58098

In #58098, when some users use the "Press this" bookmarklet, they are first taken to the _Iframed_ calypso editor e.g. `https://wordpress.com/post/$site.wordpress.com?url=foo&title=bar&text=baz`. This _iframed_ editor supports interpreting the query string parameters url, title and text. However, if for some reason the editor fails to load, the user will be taken to the wp-admin version of the editor `https://$site.wordpress.com/wp-admin/post-new.php?url=foo&title=bar&text=baz`.

This PR pulls handling of the press this query parameters into a common function that can be called from the _iframed_ calypso editor, or from the wp-admin editor.

Note: I see that "Press This" also supports the `image` and `embed` url params, but I haven't been able to get these to be passed into the URL by the bookmarklet, I have ported `image` and `embed` to the wp-admin editor with the same implementation as what was used on calypso, but I haven't yet worked out how to test either.

### Testing instructions
Changes are made to the wpcom-block-editor plugin.
Sync these changes to your sandbox with:` cd apps/wpcom-block-editor && yarn dev --sync`
Install and use the "Press This" bookmarklet as described in #58098
Copy the url you are given, replace the pattern `https://wordpress.com/post/$site.wordpress.com` with `https://$site.wordpress.com/wp-admin/post-new.php`, maintain the query string.
This simulates what happens when the calypso iframe fails to load
Ensure that press this content is loaded as expected.
